riscv/src/register/mstatus.rs

```diff
@@ -72,6 +73,20 @@ impl From<bool> for Endianness {
 }
 
 impl Mstatus {
+    /// Helper to insert a bitfield into Mstatus
+    #[inline]
+    fn bf_insert(&self, bit: usize, width: usize, val: usize) -> Self {
+        Self {
+            bits: bf_insert(self.bits, bit, width, val),
+        }
+    }
+
+    /// Helper to extract a bitfield from Mstatus
+    #[inline]
+    fn bf_extract(&self, bit: usize, width: usize) -> usize {
+        bf_extract(self.bits, bit, width)
+    }
+
```

Comment:
```suggestion
```

Reply:
I personally prefer using the new utility functions instead of filling all the registers with new methods. But it is not a hard feeling, if you think this approach has advantages, please write them so we can choose the best option.

Reply:
It's mostly to save on repeating `Self { bits: ... }` but if we're going with the `&mut self` approach there's no benefit.